examples/tiny-skia.rs

```diff
@@ -0,0 +1,211 @@
+    img.save_png("output.png").unwrap();
+}
+
+fn to_tiny_skia(color: PenikoColor) -> TinySkiaColor {
+    TinySkiaColor::from_rgba8(color.r, color.g, color.b, color.a)
+}
+
+fn render_glyph_run(glyph_run: &GlyphRun<PenikoColor>, pen: &mut TinySkiaPen<'_>, padding: u32) {
+    // Resolve properties of the GlyphRun
+    let mut run_x = glyph_run.offset();
+    let run_y = glyph_run.baseline();
+    let style = glyph_run.style();
+    let color = style.brush;
+
+    // Get the "Run" from the "GlyphRun"
+    let run = glyph_run.run();
+
+    // Resolve properties of the Run
+    let font = run.font();
+    let font_size = run.font_size();
+
+    // Get glyph outlines using Skrifa. This can be cached in production code.
+    let font_collection_ref = font.data.as_ref();
+    let font_ref = ReadFontsRef::from_index(font_collection_ref, font.index).unwrap();
+    let outlines = font_ref.outline_glyphs();
+
+    // Iterates over the glyphs in the GlyphRun
+    for glyph in glyph_run.glyphs() {
+        let glyph_x = run_x + glyph.x + padding as f32;
+        let glyph_y = run_y - glyph.y + padding as f32;
+        run_x += glyph.advance;
+
+        let glyph_id = GlyphId::from(glyph.id);
+        let glyph_outline = outlines.get(glyph_id).unwrap();
+
+        pen.set_origin(glyph_x, glyph_y);
+        pen.set_color(to_tiny_skia(color));
+        pen.draw_glyph(&glyph_outline, font_size);
+    }
+}
+
+struct TinySkiaPen<'a> {
+    pixmap: PixmapMut<'a>,
+    x: f32,
+    y: f32,
+    paint: Paint<'static>,
+    open_path: PathBuilder,
+}
+
+impl TinySkiaPen<'_> {
+    fn new(pixmap: PixmapMut) -> TinySkiaPen {
+        TinySkiaPen {
+            pixmap,
+            x: 0.0,
+            y: 0.0,
+            paint: Paint::default(),
+            open_path: PathBuilder::new(),
+        }
+    }
+
+    fn set_origin(&mut self, x: f32, y: f32) {
+        self.x = x;
+        self.y = y;
+    }
+
+    fn set_color(&mut self, color: TinySkiaColor) {
+        self.paint.set_color(color);
+    }
+
+    fn draw_glyph(&mut self, glyph: &OutlineGlyph<'_>, size: f32) {
+        let settings = DrawSettings::unhinted(Size::new(size), LocationRef::default());
+        glyph.draw(settings, self).unwrap();
+    }
+}
+
+impl OutlinePen for TinySkiaPen<'_> {
+    fn move_to(&mut self, x: f32, y: f32) {
+        self.open_path.move_to(self.x + x, self.y - y);
+    }
+
+    fn line_to(&mut self, x: f32, y: f32) {
+        self.open_path.line_to(self.x + x, self.y - y);
+    }
+
+    fn quad_to(&mut self, cx0: f32, cy0: f32, x: f32, y: f32) {
+        self.open_path
+            .quad_to(self.x + cx0, self.y - cy0, self.x + x, self.y - y);
+    }
+
+    fn curve_to(&mut self, cx0: f32, cy0: f32, cx1: f32, cy1: f32, x: f32, y: f32) {
+        self.open_path.cubic_to(
+            self.x + cx0,
+            self.y - cy0,
+            self.x + cx1,
+            self.y - cy1,
+            self.x + x,
+            self.y - y,
+        );
+    }
+
+    fn close(&mut self) {
```

Comment:
This should simply call close on the open path. The actual rendering should be done after the call to `OutlineGlyph::draw` completes.

Reply:
Ah, I tried that and the fill rule change individually but not together.

Interestingly just moving the code to that location casued a panic when unwrapping the result of `PathBuilder::finish`. But turning that unwrap into an "if let" seems to produce decent visual results even though there are ~12 invalid paths.